src/script/sign.cpp

```diff
@@ -149,6 +149,7 @@ static bool CreateTaprootScriptSig(const BaseSignatureCreator& creator, Signatur
     auto it = sigdata.taproot_script_sigs.find(lookup_key);
     if (it != sigdata.taproot_script_sigs.end()) {
         sig_out = it->second;
+        return true;
```

Comment:
9b86c58d95539056981c2a04a7345b992b1ce70c: this seems like a bug fix, is there a test you could add that demonstrates this?

Reply:
The existing tests already cover this case (try reverting this commit, the tests will fail).

Reply:
On your latest rebase f7b61d2, when I revert 9b86c58d95539056981c2a04a7345b992b1ce70c, then indeed `wallet_taproot.py` fails. Still, having the fix and test in the same commit would be much clearer. Moving this bit from the test over to this commit does the trick:

```patch
-            res = self.psbt_offline.walletprocesspsbt(psbt)
-            assert(res['complete'])
+            res = self.psbt_offline.walletprocesspsbt(psbt=psbt, finalize=False)
+
+            decoded = self.psbt_offline.decodepsbt(res["psbt"])
```

Reply:
There isn't really a specific test case for this bug; the test was written first and the bug found via the test.

Reply:
I've moved the change that causes the test failure into the previous commit.